docs/data/pages.ts

```diff
@@ -314,18 +314,18 @@ const pages: MuiPage[] = [
         pathname: '/x/react-date-pickers/localization-group',
         subheader: 'Localization',
         children: [
+          {
```

Comment:
I'm moving "Translated components" on top because the 3 other pages are related to date localization so it was weird for me to have the translation in the middle.

I also now use "Translated components" instead of "Component localization" to make it clearer that it's different from "Date localization".

Reply:
Would it then make sense to align the menu item name with the page heading? 🤔

Reply:
Totally
Updated